base/dcp_test.go

```diff
@@ -59,19 +59,32 @@ func TestTransformBucketCredentials(t *testing.T) {
 }
 
 func TestDCPKeyFilter(t *testing.T) {
-	assert.True(t, dcpKeyFilter([]byte("doc123")))
-	assert.True(t, dcpKeyFilter([]byte(UserPrefix+"user1")))
-	assert.True(t, dcpKeyFilter([]byte(RolePrefix+"role2")))
-	assert.True(t, dcpKeyFilter([]byte(UnusedSeqPrefix+"1234")))
-	assert.True(t, dcpKeyFilter([]byte(SGCfgPrefixWithGroupID("")+"123")))
-	assert.True(t, dcpKeyFilter([]byte(SGCfgPrefixWithGroupID("group")+"123")))
-
-	assert.False(t, dcpKeyFilter([]byte(SyncSeqKey)))
-	assert.False(t, dcpKeyFilter([]byte(SyncDocPrefix+"unusualSeq")))
-	assert.False(t, dcpKeyFilter([]byte(SyncFunctionKeyWithoutGroupID)))
-	assert.False(t, dcpKeyFilter([]byte(DCPCheckpointPrefixWithoutGroupID+"12")))
-	assert.False(t, dcpKeyFilter([]byte(TxnPrefix+"atrData")))
-	assert.False(t, dcpKeyFilter([]byte(DCPCheckpointPrefixWithGroupID("group")+"12")))
+
+	testCases := []struct {
+		name     string
+		metaKeys *MetadataKeys
+	}{
+		{"default meta keys", DefaultMetadataKeys},
+		{"db specific meta keys", NewMetadataKeys("dbname")},
```

Comment:
Could give us ever so slightly more coverage of the function used by `NewDatabaseContext`

```suggestion
		{"default meta keys", NewMetadataKeys("")},
		{"db specific meta keys", NewMetadataKeys("dbname")},
```